nearcore/src/lib.rs

```diff
@@ -274,7 +274,12 @@ pub fn start_with_config_and_synchronization(
     );
     shards_manager_adapter.bind(shards_manager_actor);
 
-    let state_sync_dump_handle = spawn_state_sync_dump(&config, chain_genesis, runtime)?;
+    let state_sync_dump_handle = spawn_state_sync_dump(
+        &config.client_config,
+        chain_genesis,
+        runtime,
+        config.validator_signer.as_ref().map(|signer| signer.validator_id().clone()),
```

Comment:
why is this argument added?

Reply:
To know which shards exactly are being tracked.
It makes a difference if the node is a chunk only producer (COP) that tracks no shards, and is assigned to produce chunks for some of the shards. Then it will be able to dump state of those shards.

It's an exotic case - somebody runs a COP **and** also dumps state to S3.
Let me know if you think this should be skipped for simplicity.

Reply:
`It makes a difference if the node is a chunk only producer (COP) that tracks no shards, and is assigned to produce chunks for some of the shards. Then it will be able to dump state of those shards.` -- meaning the node should be able to dump state of the shard that it will start tracking for the current epoch because of catchup?

Reply:
Yes, exactly